### ISSUE TITLE
[SERVER CRASH] init.lua:93 attempt to index a nil value

### ISSUE THREAD

**Issue by insanity54** [insanity54/gemwarp#4]
```
2022-11-24 11:20:32: ERROR[Main]: ServerError: AsyncErr: Lua: Runtime error from mod 'gemwarp' in callback on_playerReceiveFields(): /home/minetest/.minetest/mods/gemwarp/init.lua:93: attempt to index a nil value
2022-11-24 11:20:32: ERROR[Main]: stack traceback:
2022-11-24 11:20:32: ERROR[Main]: 	/home/minetest/.minetest/mods/gemwarp/init.lua:93: in function </home/minetest/.minetest/mods/gemwarp/init.lua:84>
2022-11-24 11:20:32: ERROR[Main]: 	/usr/local/share/minetest/builtin/game/register.lua:431: in function </usr/local/share/minetest/builtin/game/register.lua:417>
```

**Comment by insanity54** [comment]
Third time I've seen a crash from this. Previous bugfixes are invalid? Or is the mod not up-to-date with latest patch?

```
[127391]: 2022-12-12 12:25:13: ERROR[Main]: ServerError: AsyncErr: Lua: Runtime error from mod 'gemwarp' in callback on_playerReceiveFields(): /home/minetest/.minetest/mods/gemwarp/init.lua:96: attempt >
[127391]: 2022-12-12 12:25:13: ERROR[Main]: stack traceback:
[127391]: 2022-12-12 12:25:13: ERROR[Main]:         /home/minetest/.minetest/mods/gemwarp/init.lua:96: in function </home/minetest/.minetest/mods/gemwarp/init.lua:85>
[127391]: 2022-12-12 12:25:13: ERROR[Main]:         /usr/local/share/minetest/builtin/game/register.lua:431: in function </usr/local/share/minetest/builtin/game/register.lua:417>
netest.service: Main process exited, code=exited, status=1/FAILURE
```

It seems to be caused by a new player. Maybe they lack waypoints and they somehow attempt a warp?

EDIT: OK I got reproduction! I checked the db for the player who seemed to have caused the crash.

I created a new account and was able to crash it myself when I pressed the gemwarp button.

It's caused when a player has the following metadata for ui_waypoints

```
{"data":[null,null,null,null,{"display_pos":true}],"selected":5}
```

So that means the player didn't set a waypoint position, but they did toggle the switch that displays the waypoint.

**Comment by insanity54** [comment]
fixed in v0.4.1
